### ISSUE TITLE
Codesign Error with GiphyUISDK 2.1.6

### ISSUE THREAD

**Issue by ekurutepe** [Giphy/giphy-ios-sdk#149]
We import the SDK through SPM. Everything works as expected up to 2.1.5

Today I upgraded to 2.1.6 and it broke our release pipeline with the following error when codesigning before upload: 
```
Failed to generate distribution items with error: Error Domain=DVTMachOErrorDomain Code=0 "Found an unexpected Mach-O header code: 0x72613c21" UserInfo={NSLocalizedDescription=Found an unexpected Mach-O header code: 0x72613c21, NSLocalizedRecoverySuggestion=}
```

Downgrading back to 2.1.5 fixes the issue.

**Comment by davidsiu** [comment]
Same here. It is failing on our M1 Mac Mini building in Rosetta mode.

**Comment by AndrewSB** [comment]
oh wow... I spent like 2 hours trying to figure this out, didn't realize it was Giphy. downgrading to 2.1.5 now...

https://github.com/firebase/firebase-ios-sdk/issues/6472 seems relevant.

**Comment by cgmaier** [comment]
Sorry about the delay on this - sorry to hear it seems to have led to some painful debugging : / 

I'm now back in business however, and just pushed a version `2.1.7` which should hopefully resolve this issue. I think I accidentally touched the mach-o setting in the last release. 

Please let me know if this gets things working again. thanks!

**Comment by jnwagstaff** [comment]
@cgmaier I have the same problem on `2.1.7`, `2.1.5` is the latest version that is working for me

**Comment by cgmaier** [comment]
thanks for confirming @jnwagstaff and thanks for sharing that firebase issue link @AndrewSB 

I've tried a number of things and am yet to be able to reproduce the error uploading the example app to App Store Connect. I'm using Xcode 12.4 on an M1 MacBook Air. Any ideas?

I can't seem to determine what may be different about release `2.1.5` - the build script and build settings don't appear to have been modified. 

I created this [branch](https://github.com/Giphy/giphy-ios-sdk/tree/spm-upload-test) to test SPM (rather than CocoaPods) in the swift Example app, where upload works for me. feel free to plug in your own signing credentials and a bundle id to the project in order to test archive / upload. 

also wondering there is anything in `Other Linker Flags` which may be responsible? 

thanks in advance for your helping to debug. hopefully we can resolve this issue soon! 

cc @ekurutepe @davidsiu

**Comment by cgmaier** [comment]
hey all,

we've determined this was an issue with Xcode, as noted in their [12.4 release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-12_4-release-notes):

> Swift Packages

> Known Issues

> If you use a Swift package with binary dependencies in an app with extensions, the build system incorrectly embeds the binary dependencies alongside the extension in the PlugIns directory, causing validation of the archived app to fail. (69834549) (FB8761306)

> Workaround: Add a scheme post-build action which removes the embedded binaries from the PlugIns directory after the build, e.g. rm -rf "${TARGET_BUILD_DIR}/${TARGET_NAME}.app"/PlugIns/*.framework.
> 

this appears to be fixed with [Xcode 12.5](https://developer.apple.com/documentation/xcode-release-notes/xcode-12_5-release-notes): 

> The Swift Package Manager now builds package products and targets as dynamic frameworks automatically, if doing so avoids duplication of library code at runtime. (59931771) (FB7608638)

cc @ALexanderLonsky
